### ISSUE TITLE
Custom SD pipeline with hard-coded left truncation of text prompts 

### ISSUE THREAD

**Issue by avipartho** [kohjingyu/gill#4]
Hi, 

Thanks a lot for sharing your codebase. As I was trying to run your code on my own data, I figured out that [here](https://github.com/kohjingyu/gill/blob/51f3d98a2b3a0d9776e221cf3c3b5331326c0dc4/gill/custom_sd.py#L266), you have set `truncate_side` to be `left`. What is the rationale behind it? I can see that the [official SD pipeline](https://github.com/huggingface/diffusers/blob/5439e917cacc885c0ac39dda1b8af12258e6e16d/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L356) truncates text prompts from the right side. Truncating from the left leaves me with a series of `<|endoftext|>` (as expected). It would be great if you could take a look. Thanks.

**Comment by kohjingyu** [comment]
Thanks for pointing this out! (I'm assuming that you tried to run the preprocessing pipeline)

It should indeed be `truncate_side = 'right'` for the preprocessing pipeline, which I've just updated in the code. For running GILL, you shouldn't need to touch the `_encode_prompts()` function, so performance of the pretrained models should be unaffected by this.

The only time it should be set to `truncate_side = 'left'` (and why it was originally as such in the code) is when we run SD on the VIST and VisDial evals. Many of these examples are longer than 77 tokens, and SD misses most of the relevant context if we truncate it from the left. Hence, we truncated from the left in the baseline results we report in our paper, so that it gets most of the recent context (rather than the starting text).

Hope that make sense!

**Comment by avipartho** [comment]
Thanks for your prompt response! Yes, I was running the script to preprocess my own data. 

Now with this recent change in the code, wouldn't it be problematic for evaluation? Is it possible to turn it into an argument that the user can pass? Huggingface tokenizer class `PreTrainedTokenizer` indeed has a truncation_side parameter ([ref](https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizer.truncation_side)) and `CLIPTokenizer` inherits `PreTrainedTokenizer` ([source code](https://github.com/huggingface/transformers/blob/66ded238cd04e29ba98485984dd647e7d37d1603/src/transformers/models/clip/tokenization_clip.py#L260)).

**Comment by kohjingyu** [comment]
That's a fair point. I tried setting that parameter previously, but it didn't work correctly (it just truncated left regardless), which is why I implemented it as such in `custom_sd.py` rather than just setting the tokenizer arguments. They might have fixed it since then, but I haven't checked (so I'm opting to keep the current implementation)

I've updated the model to accept `truncation_side` as an init argument, with it defaulting to 'right': https://github.com/kohjingyu/gill/commit/c1d33098fba2e5dc30f3f940449af1469ce549ab

Thanks again for spotting this -- let me know if you have any other concerns!
